packages/@o3r/localization/schematics/add-localization-key/index.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  chain,
+  noop,
+  Rule,
+  SchematicContext,
+  Tree
+} from '@angular-devkit/schematics';
+import {
+  applyEsLintFix,
+  getO3rComponentInfo,
+  isO3rClassComponent
+} from '@o3r/schematics';
+import { dirname, resolve } from 'node:path';
+import * as ts from 'typescript';
+import type { NgAddLocalizationKeySchematicsSchema } from './schema';
+
+const getLocalizationInformation = (componentPath: string, tree: Tree) => {
+  const componentSourceFile = ts.createSourceFile(
+    componentPath,
+    tree.readText(componentPath),
+    ts.ScriptTarget.ES2015,
+    true
+  );
+
+  const o3rClass = componentSourceFile.statements.find((statement): statement is ts.ClassDeclaration => ts.isClassDeclaration(statement) && isO3rClassComponent(statement))!;
+  let localizationJsonPath: string | undefined;
+  let defaultTranslationVariableName: string | undefined;
+  let translationsVariableName: string | undefined;
+  let translationsVariableType: string | undefined;
+  o3rClass.members.forEach((member) => {
+    if (ts.isPropertyDeclaration(member) && ts.isIdentifier(member.name) && member.type && ts.isTypeReferenceNode(member.type) && ts.isIdentifier(member.type.typeName)) {
+      const decorators = [...(ts.getDecorators(member) || [])];
+      const localizationDecorator = decorators.find((decorator): decorator is ts.Decorator & { expression: ts.CallExpression & { arguments: [ts.StringLiteral]} } =>
+        ts.isCallExpression(decorator.expression)
+        && ts.isIdentifier(decorator.expression.expression)
+        && decorator.expression.expression.escapedText.toString() === 'Localization'
+        && ts.isStringLiteral(decorator.expression.arguments[0])
+      );
+      if (localizationDecorator) {
+        localizationJsonPath = resolve(dirname(componentPath), localizationDecorator.expression.arguments[0].text);
+        translationsVariableName = member.name.escapedText.toString();
+        translationsVariableType = member.type.typeName.escapedText.toString();
+        if (member.initializer && ts.isIdentifier(member.initializer)) {
+          defaultTranslationVariableName = member.initializer.escapedText.toString();
+        }
+      }
+    }
+  });
+
+  if (!localizationJsonPath || !tree.exists(localizationJsonPath)) {
+    throw new Error(
```

Comment:
Actually, the coolest would be if this schematics will ask you if you want to run the `localization-to-component` (and run it for you)

Reply:
as discussed will be done in another PR

Reply:
https://github.com/AmadeusITGroup/otter/issues/480